fbw-a380x/src/systems/instruments/src/MsfsAvionicsCommon/gauges.tsx

```diff
@@ -390,3 +406,255 @@ export class GaugeComponent extends DisplayComponent<GaugeComponentProps> {
+            x: this.props.x + (valueMaxDir.x * this.props.radius),
+            y: this.props.y + (valueMaxDir.y * this.props.radius),
+        };
+
+        this.thrustPath.set([
+            `M ${this.props.x},${this.props.y} L ${valueIdleEnd.x},${valueIdleEnd.y}`,
+            `A ${this.props.radius} ${this.props.radius} 0 ${this.props.reverse ? '0' : '1'} 1 ${valueMaxEnd.x} ${valueMaxEnd.y}`,
+            `M ${valueMaxEnd.x} ${valueMaxEnd.y} L ${this.props.x},${this.props.y}`,
+        ].join(' '));
+    }
+
+    render(): VNode {
+        return (
+            <>
+                <g class='GaugeComponent'>
+                    <g class={this.props.visible.map((it) => it ? 'Show' : 'Hide')}>
+                        <path d={this.thrustPath} class={this.props.class} />
+                    </g>
+                </g>
+            </>
+        );
+    }
+}
+
+interface ThrustTransientComponentProps {
+    x: number;
+    y: number;
+    min: number;
+    max: number;
+    radius: number;
+    startAngle: Subscribable<number>;
+    endAngle: Subscribable<number>;
+    class: string;
+    visible: Subscribable<boolean>;
+    thrustActual: Subscribable<number>;
+    thrustTarget: Subscribable<number>;
+}
+export class ThrustTransientComponent extends DisplayComponent<ThrustTransientComponentProps> {
+    private startAngle: number = 0;
+
+    private endAngle: number = 0;
+
+    private thrustActual: number = 0;
+
+    private thrustTarget: number = 0;
+
+    private sweep1 = Subject.create('');
+    private sweep2 = Subject.create('');
+    private sweep3 = Subject.create('');
+    private sweep4 = Subject.create('');
+
+    private endPath = Subject.create('');
+
+    constructor(props: ThrustTransientComponentProps) {
+        super(props);
+
+        this.startAngle = this.props.startAngle.get();
+        this.endAngle = this.props.endAngle.get();
+        this.update();
+    }
+
+    private sweepHorizontal(radius: number) {
+        const valueIdleDir = valueRadianAngleConverter(this.thrustActual, this.props.min, this.props.max, this.endAngle, this.startAngle);
+        const valueIdleEnd = {
+            x: this.props.x + (valueIdleDir.x * radius),
+            y: this.props.y + (valueIdleDir.y * radius),
+        };
+        const valueMaxDir = valueRadianAngleConverter(this.thrustTarget, this.props.min, this.props.max, this.endAngle, this.startAngle);
+        const valueMaxEnd = {
+            x: this.props.x + (valueMaxDir.x * radius),
+            y: this.props.y + (valueMaxDir.y * radius),
+        };
+
+        const diffAngle = valueMaxDir.angle - valueIdleDir.angle;
+
+        return [
+            `M ${valueIdleEnd.x},${valueIdleEnd.y}`,
+            `A ${radius} ${radius} 0 ${Math.abs(diffAngle) > 180 ? '1' : '0'} ${diffAngle < 0 ? '0' : '1'} ${valueMaxEnd.x} ${valueMaxEnd.y}`,
+        ].join(' ');
+    }
+
+    onAfterRender(node: VNode): void {
+        super.onAfterRender(node);
+
+        this.props.startAngle.sub((a) => {
```

Comment:
Are `startAngle` and `endAngle` actually subjects? From the usages I could see their values are always static. If that'S the case, would be good to actually represent them as nubmers

Reply:
I thought I would need it for the ACUTE thrust limits, but it doesn't seem that way. So I changed it back to simple numbers.